tests/test_signals/test_ebsd_master_pattern.py

```diff
@@ -758,3 +775,487 @@ def test_vector2xy(self):
         ]
         assert np.allclose(_vector2lambert.py_func(xyz), lambert_xy)
         assert np.allclose(_vector2lambert(xyz), lambert_xy)
+
+
+class TestFitPatternDetectorOrientation:
+    """
+    Test the fit between an EBSD pattern generated
+    from a master pattern and an associated
+    GeometricalKikuchiPatternSimulation for different
+    detector orientations.
+    """
+
+    detector = kp.detectors.EBSDDetector(
+        shape=(480, 640), px_size=50, pc=(20, 20, 15000), convention="emsoft4"
+    )
+
+    phase = kp.data.nickel_ebsd_master_pattern_small().phase
+
+    pqr_p = [(1, 1, 1), (2, 0, 0), (2, 2, 0), (3, 1, 1)]
+
+    ref = ReciprocalLatticeVector(phase=phase, hkl=pqr_p)
+    ref = ref.symmetrise()
+
+    simulator = kp.simulations.KikuchiPatternSimulator(ref)
+
+    rotations = Rotation.from_euler([23, 14, 5], degrees=True)
+
+    # Transformation from CSs to cartesian crystal reference frame CSc
+    u_o = rotations.to_matrix().squeeze()
+
+    # Transformation from CSc to direct crystal reference frame CSk
+    u_a = phase.structure.lattice.base
+
+    def setup_detector_sim_and_u_os(self, tilt, azimuthal, euler_2):
+        """Setup the detector with the correct tilt, azimuthal and
```

Comment:
I'd consider this docstring to be superfluous as the code is easy to read and self-documenting. I'd remove it.